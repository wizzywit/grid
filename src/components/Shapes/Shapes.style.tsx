import styled from "styled-components";

export const Styles = {
  ShapesContainer: styled.div`
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    /* justify-content: space-between;
    width: 50%; */
    gap: 25px;
    margin-bottom: 50px;
  `,
};
