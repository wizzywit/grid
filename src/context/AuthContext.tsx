import React from "react";

interface DataArray {
  color: string;
  shape: string;
}

export type Filter = (
  c: Array<string> | ((c: Array<string>) => Array<string>)
) => void;

type PerformFilterType = (
  shape: string,
  shapesFilter: Array<string> | undefined,
  filter: Filter | undefined,
  type: string
) => void;

type AuthContextType = {
  login?: () => void;
  logout?: () => void;
  data?: Array<DataArray>;
  shapesFilter?: Array<string>;
  setShapesFilter?: Filter;
  colorsFilter?: Array<string>;
  setColorsFilter?: Filter;
  isLoggedIn?: boolean;
  performSelect?: PerformFilterType;
};
const AuthContext = React.createContext<AuthContextType>({});

export default AuthContext;
