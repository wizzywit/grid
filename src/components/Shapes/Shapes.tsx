import React, { useContext } from "react";
import { Styles } from "./Shapes.style";
import ShapeButton from "../ShapeButton/ShapeButton";
import AuthContext from "../../context/AuthContext";

const Shapes = () => {
  const { shapesFilter, shapes } = useContext(AuthContext);

  return (
    <Styles.ShapesContainer>
      {shapes?.map((shape) => (
        <ShapeButton
          key={shape}
          text={shape}
          isActive={shapesFilter?.includes(shape.toLowerCase())}
        />
      ))}
    </Styles.ShapesContainer>
  );
};

export default Shapes;
