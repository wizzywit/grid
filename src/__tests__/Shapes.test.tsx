import React from "react";

import { render } from "@testing-library/react";

import Shapes from "../components/Shapes/Shapes";
import AuthContext from "../context/AuthContext";

const shapes = ["Oval", "Traingle", "Round", "Square", "Rectangle"];

describe("Shapes", () => {
  test("should display list of colors", async () => {
    const { container, findAllByTestId } = render(
      <AuthContext.Provider value={{ shapes }}>
        <Shapes />
      </AuthContext.Provider>
    );
    expect(container).toMatchSnapshot();
    const children = await findAllByTestId("shape");
    expect(children).toHaveLength(shapes.length);
  });
});
