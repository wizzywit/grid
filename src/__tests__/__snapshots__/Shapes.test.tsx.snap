// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shapes should display list of colors 1`] = `
<div>
  <div
    class="sc-bdnxRM gYskiT"
  >
    <button
      class="sc-gtsrHT icDXbM"
      data-testid="shape"
    >
      <p>
        Oval
      </p>
    </button>
    <button
      class="sc-gtsrHT icDXbM"
      data-testid="shape"
    >
      <p>
        Traingle
      </p>
    </button>
    <button
      class="sc-gtsrHT icDXbM"
      data-testid="shape"
    >
      <p>
        Round
      </p>
    </button>
    <button
      class="sc-gtsrHT icDXbM"
      data-testid="shape"
    >
      <p>
        Square
      </p>
    </button>
    <button
      class="sc-gtsrHT icDXbM"
      data-testid="shape"
    >
      <p>
        Rectangle
      </p>
    </button>
  </div>
</div>
`;
